### ISSUE TITLE
Allow setting individual thickness and length of lines, maybe directly or using a "prority number"

### ISSUE THREAD

**Issue by 56independent** [vasturiano/force-graph#317]
**Is your feature request related to a problem? Please describe.**
I made an [issue](https://github.com/zadam/trilium/issues/3840) on Trilium asking for the graph to be able to contract or pull individual links, keeping porportion similar. However, the developer discussed the fact that this library limits this from happening.

> One problem with that is that IIRC the library which I'm using to render the maps ("force-graph") does not support per-connection "strength". They all "pull" the notes together with the same force.

**Describe the solution you'd like**

I would like the ability to set the thickness and length of lines in two ways:

* Setting an individual `length:` and `thickness:` (measured in arbitary units) to show how they're displayed, where in "impossible situations" (such as three nodes connected by 3 2-length lines), the biggest number is made a little bigger and the smallest a little smaller, the node with the most connections having priority
* A priority number that works as above, but automatically determines the thickness

I don't have much experience with this project (yet), but i think that in the below snippet of [code](https://github.com/vasturiano/force-graph/blob/master/example/basic/index.html#L14):

```
    const gData = {
      nodes: [...Array(N).keys()].map(i => ({ id: i })),
      links: [...Array(N).keys()]
        .filter(id => id)
        .map(id => ({
          source: id,
          target: Math.round(Math.random() * (id-1))
        }))
    };
```

we could add line thicknesses like so?

```
    const gData = {
      nodes: [...Array(N).keys()].map(i => ({ id: i })),
      links: [...Array(N).keys()]
        .filter(id => id)
        .map(id => ({
          source: id,
          target: Math.round(Math.random() * (id-1))
        .looks(id => ({
          length: 20,
          thickness: 10
        })
        }))
    };
```

**Comment by vasturiano** [comment]
@56independent thanks for reaching out.

It's certainly possible to add line thickness and calibrate the link attraction force between nodes.

Let's look at line thickness first. This is a purely cosmetic item (does not influence the force engine), and you can manipulate it per link, using the `.lineWidth` method.

As for the distance, or any other attributes of the link attraction force, you can calibrate it using all settings found in [d3-force forceLink](https://github.com/d3/d3-force#links). The link force works a little bit like a spring in which each link has a target length (at rest), and any push/pulls from that position will be counter-acted with a linear attraction force. In this case if you're interested in changing the target distance of the links, you can do something like:
```javascript
myGraph.d3Force('link').distance(link => /* your code */)
```

The same way you can modify other attributes of the force, like **strength** for instance.

**Comment by 56independent** [comment]
Ok, so that covers the first method. However, i would like to see a "priority" value as well, which abstracts away the need for both to be modified; if the actual length post-being-made-a-spring is shorter then the amount set, it then makes the line thicker, and vice versa. 

See my potential algorithm:

1. Find the set priority number 
2. Multiply it by the "length factor" constant
3. Use this to set the length
4. Find the actual length afterwards
5. Use the difference and multiply it by the "thickness reaction constant" to get the thickness
6. Set the thickness to that amount

This then allows us to see how the nodes have been forced to change and where the most extreme relationships are.

**Comment by vasturiano** [comment]
> Are you sure the link length is individually settable? I'd like each link to have it's own target distance.

Yes, both the `distance` and `strength` of the link force can be set independently per link, as mentioned in the docs.

This logic is quite app specific, so it makes sense for it to exist in the app, not in this module. But you should have all the configuration available from this module to be able to build it.
